tutorials/data-science-preprocessing/index.md

```diff
@@ -0,0 +1,437 @@
+            schema=', '.join([
+                'fall:STRING',
+                'year:INTEGER',
+                'nametype:STRING',
+                'mass:FLOAT',
+                'name:STRING',
+                'class:STRING',
+                'latitude:FLOAT',
+                'longitude:FLOAT',
+                'id:STRING',
+            ]),
+            create_disposition=(
+                apache_beam.io.BigQueryDisposition.CREATE_IF_NEEDED),
+            write_disposition=(
+                apache_beam.io.BigQueryDisposition.WRITE_TRUNCATE))))
+
+    p.run()
+```
+
+Note that this example takes advantage of the built-in [`BigQuerySink`][bq-sink]
+to output the result into BigQuery, to analyze later.
+
+[bq-sink]: https://github.com/apache/incubator-beam/tree/python-sdk/sdks/python#bigquery
+
+## Run the data cleansing pipeline
+
+A great thing about Apache Beam and Cloud Dataflow is that you can run it
+locally for sample datasets, or datasets that are otherwise relatively small.
+But then you can take the same code, add a couple extra flags, and run it using
+Cloud Dataflow which can handle orders of magnitude more data at a time.
+
+Because this sample uses a `BigQuerySink` for its output, you must first perform
+some setup:
+
+* First, make sure you're using the correct project:
+
+      $ gcloud config set project your-project-id
+      Updated property [core/project]
+      $
+
+* Now, create a BigQuery dataset using the `bq` command provided by the Google
+  Cloud SDK:
+
+      $ bq mk meteor_dataset
+      Dataset 'your-project-id:meteor_dataset' successfully created
+      $
+
+* To ensure that this example's dependencies don't conflict with any existing
+  libraries on your system, create a virtualenv and install the script
+  dependencies:
+
+      $ virtualenv venv
+      ...
+      $ source venv/bin/activate
+      (venv) $ pip install -r requirements.txt
+      ...
+      (venv) $
+
+With the BigQuery dataset created and your requirements installed, you are ready
+to run your pipeline.
+
+### Running locally
+
+    python clean.py meteors.json your-project-id:meteor_dataset.cleansed
+
+### Running on Cloud Dataflow
+
+To run the script on Cloud Dataflow, you'll have to specify your project, and a
+bucket in Google Cloud Storage:
+
+    PROJECT=your-project-id
+    # The $PROJECT.appspot.com bucket is created automatically with your project
+    BUCKET=gs://$PROJECT.appspot.com
+
+Now upload the json data file to Cloud Storage using the `gsutil` tool included
+in the Google Cloud SDK:
+
+    gsutil cp meteors.json gs://$BUCKET/
+
+Then run:
+
+    python clean.py gs://$BUCKET/meteors.json $PROJECT:meteor_dataset.cleansed \
+        --project $PROJECT \
+        --job_name $PROJECT-meteors \
+        --runner BlockingDataflowPipelineRunner \
+        --staging_location $BUCKET/staging \
+        --temp_location $BUCKET/temp \
+        --output $BUCKET/output
+
+You can check the status of your job on the [Dataflow
```

Comment:
Having a link split across two lines breaks it. For long links, we typically break the line at the beginning of the link text and let that one line flow past the 80-character mark.